homeassistant/util/__init__.py

```diff
@@ -32,6 +33,27 @@
 RE_SANITIZE_PATH = re.compile(r"(~|\.(\.)+)")
 
 
+def raise_if_invalid_filename(filename: str) -> None:
+    """
+    Check if a filename is valid.
+
+    Raises a ValueError if the filename is invalid.
+    """
+    if RE_SANITIZE_FILENAME.sub("", filename) != filename:
```

Comment:
Maybe use `.search()` here, that seems clearer?

Reply:
from a performance aspect if looks like `.match()` is the winner, this is with 10.000 checks:
```
--- 0.004220485687255859 seconds for 'sub'---
--- 0.002567291259765625 seconds for 'search'---
--- 0.0024166107177734375 seconds for 'match'---
```